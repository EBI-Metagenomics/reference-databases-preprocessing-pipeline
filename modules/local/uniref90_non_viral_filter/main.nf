process UNIREF90_NON_VIRAL_FILTER {
    label 'process_single'
    container 'community.wave.seqera.io/library/pip_biopython_pyfastx_taxoniq:c440284a61b91ed0'

    input:
    path uniref90_fasta

    output:
    path '*.filtered.fasta', emit: filtered_proteins
    path '*.protid2taxid'  , emit: protid2taxid
    path 'versions.yml'    , emit: versions

    script:
    """
    uniref90_non_viral_filter.py ${uniref90_fasta} uniref90

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
        biopython: \$(python -c "import importlib.metadata; print(importlib.metadata.version('biopython'))")
        taxoniq: \$(python -c "import importlib.metadata; print(importlib.metadata.version('taxoniq'))")
        uniref90_db: ${params.uniref90_version}
    END_VERSIONS
    """
}