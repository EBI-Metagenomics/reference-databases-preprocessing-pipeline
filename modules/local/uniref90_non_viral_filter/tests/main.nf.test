nextflow_process {

    name "Test Process UNIREF90_NON_VIRAL_FILTER"
    script "../main.nf"
    process "UNIREF90_NON_VIRAL_FILTER"

    test("Remove viral proteins from protein fasta using TaxID field") {

        when {
            process {
                """
                input[0] = Channel.fromPath("${moduleDir}/tests/data/uniref90.test.fasta", checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
