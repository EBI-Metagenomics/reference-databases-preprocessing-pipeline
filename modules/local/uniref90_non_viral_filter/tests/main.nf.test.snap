{
    "Remove viral proteins from protein fasta using TaxID field": {
        "content": [
            {
                "0": [
                    "uniref90_non_viral.fasta:md5,e5ee4d75d12e9dd95fd7f7ffe9f32208"
                ],
                "filtered_fasta": [
                    "uniref90_non_viral.fasta:md5,e5ee4d75d12e9dd95fd7f7ffe9f32208"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-06T12:08:11.280031"
    }
}